src/nvim/getchar.c

```diff
@@ -3447,6 +3513,11 @@ static void showmap(mapblock_T *mp, bool local)
     msg_outtrans_special(s, false, 0);
     xfree(s);
   }
+
+  if (mp->m_desc != NULL) {
+    msg_puts("\n                 ");  // Shift line to same level as rhs.
+    msg_puts(mp->m_desc);
```

Comment:
Not sure about putting `desc` on a newline. If plugins or scripts depend on there being one mapping per line this would break that.

Knowing the number of the Lua ref is (imo) not that useful or interesting. What if instead we indicate a mapping uses a Lua function by prepending it with an `L` or `#` (similar to how buffer-local mappings have a `@` and non-remappable mappings use `*`). So e.g. `:map` would show

```
n  ZT          L Open a new tab
```

This should be pretty straightforward to add. The relevant code is at line 3486 in `getchar.c` and the docs under `|map-listing|`.

Reply:
But non lua keymaps can also have description . And lua keymap without description has to be shown as ref no any way . description being on separate line personally seems nicer.

Also I can't see why someone would be parsing `:map` command output when `maparg()` exists ?

Reply:
>Also I can't see why someone would be parsing :map command output when maparg() & nvim_get_keymap exists ?

Just because it doesn't make sense doesn't mean people don't do it :) Though I'm not sure anyone actually is doing it, so maybe my concerns aren't valid.

But you're right, `desc` can be provided for non-Lua mappings as well, so this is probably fine.